File: src/main/java/lsd/format/json/JsonParser.java
package lsd.format.json;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.NoArgsConstructor;
import lombok.extern.slf4j.Slf4j;

import java.util.Map;
import java.util.Optional;

import static java.util.Optional.empty;
import static lombok.AccessLevel.PRIVATE;
import static org.apache.commons.lang3.StringUtils.isBlank;

@Slf4j
@NoArgsConstructor(access = PRIVATE)
public class JsonParser {

    private final static ObjectMapper objectMapper = ObjectMapperCreator.create();

    public static Optional<Map<String, Object>> indentJson(final String document) {
        if (isBlank(document)) {
            return empty();
        }

        try {
            return Optional.of(objectMapper.readValue(document, new TypeReference<>() {}));
        } catch (final JsonProcessingException e) {
            log.trace(e.getMessage());
            return empty();
        }
    }
}

File: src/main/java/lsd/format/json/JsonParser.kt
package lsd.format.json

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.core.type.TypeReference
import lsd.format.log

private val objectMapper = createObjectMapper()

fun convert(document: String?): Map<String, Any>? {
    return document?.let {
        try {
            objectMapper.readValue(it, object : TypeReference<Map<String, Any>>() {})
        } catch (e: JsonProcessingException) {
            log().trace(e.message)
            null
        }
    }
}
